### ISSUE TITLE
Conflicting ChatRequest name

### ISSUE THREAD

**Issue by agoncal** [Azure-Samples/azure-openai-rag-workshop-java#20]
In the workshop there is a `ChatRequest` class. But in the last LC4j versions there is now an experimental `ChatRequest` API too. And if you leave your IDE doing the auto-import, it imports the LC4j one. We should change the name of the API (and also the `ChatResource`....). Maybe `ChatRagRequest` or something like that

https://github.com/langchain4j/langchain4j/blob/main/langchain4j-core/src/main/java/dev/langchain4j/model/chat/request/ChatRequest.java

**Comment by jdubois** [comment]
Yes we should refactor it to something meaningful for us, `AssistantChatRequest`?

**Comment by agoncal** [comment]
Yes, we should rename all our classes so we don't have any conflict and the naming is consistent. What about having:

* `AssistantChatMessage`
* `AssistantChatRequest`
* `AssistantChatResource`
* `AssistantChatResponse`

**Comment by jdubois** [comment]
Oh yes!! Can you do it?

**Comment by agoncal** [comment]
Yes, I'll do it, but after our workshop tomorrow ;o)
